skimage/restoration/deconvolution.py

```diff
@@ -357,36 +368,129 @@ def richardson_lucy(image, psf, iterations=50, clip=True):
     >>> camera += 0.1 * camera.std() * np.random.standard_normal(camera.shape)
     >>> deconvolved = restoration.richardson_lucy(camera, psf, 5)
 
+    Notes
+    -----
+    The Richardson-Lucy algorithm is an iterative algorithm to
+    deconvolve an image using a given point spread function (PSF).
+
+    If no PSF is provided, the algorithm performs an "inverse" Richardson
+    Lucy algorithm as described in Fish et al., 1995.
+    It is an iterative process where the PSF
+    and image is deconvolved, respectively.
+    It is more noise tolerant than other algorithms,
+    such as Ayers-Dainty and the Wiener filter algorithms
+    (taken from the paper).
+
+    The algorithm performs well with gaussian PSFs and can recover
+    them nicely without any prior knowledge. If one has already an
+    educated guess, one should pass the PSF as argument to the function.
+    Note, that the PSF should have the same shape as the image,
+    and the PSF should be centered.
+
+    Due to its nature, the algorithm may divide by 0.
+    The function catches this issue and aborts the iterative process.
+    Mostly, the optimal number of iterations is before this error may occur.
+
     References
     ----------
     .. [1] https://en.wikipedia.org/wiki/Richardson%E2%80%93Lucy_deconvolution
+
+    .. [2] Fish, D. A., A. M. Brinicombe, E. R. Pike, and J. G. Walker.
+           "Blind deconvolution by means of the Richardson–Lucy algorithm."
+           JOSA A 12, no. 1 (1995): 58-65. doi: 10.1364/JOSAA.12.000058.
     """
+    # Set the time aspect of the PSF depending on the algorithm used.
+    if psf is None:
+        time_psf = image.shape
+    else:
+        time_psf = psf.shape
+
     # compute the times for direct convolution and the fft method. The fft is of
     # complexity O(N log(N)) for each dimension and the direct method does
     # straight arithmetic (and is O(n*k) to add n elements k times)
-    direct_time = np.prod(image.shape + psf.shape)
-    fft_time =  np.sum([n*np.log(n) for n in image.shape + psf.shape])
+    direct_time = np.prod(image.shape + time_psf)
+    fft_time = np.sum([n * np.log(n) for n in image.shape + time_psf])
 
     # see whether the fourier transform convolution method or the direct
-    # convolution method is faster (discussed in scikit-image PR #1792)
+    # convolution method is faster (discussed in scikit-image PR #1792).
     time_ratio = 40.032 * fft_time / direct_time
```

Comment:
The code I ran is in this gist: https://gist.github.com/stsievert/01ba818e2b16fcdec34d, and this is the comment shows the plots it generates: https://github.com/scikit-image/scikit-image/pull/1792#issuecomment-163809219. I think a straightforward rerunning of this code should be sufficient if time is spent.

I did revisit this for the upstream merge, https://github.com/scipy/scipy/pull/5608. I spent a significant amount of time tuning the ratios<sup>1</sup>. There's a function called [`_fftconv_faster`][1] which has updated ratios, but I'm not sure if it directly applies. That function notes which computer I ran it on, but I'm not sure that's the same computer I ran the code simulations on.

Thanks for this comment; I'll try to revisit this. Plus, blind deconvolution is the exact issue where I ran into the problem that motivated #1792.

<sup>1: because I had to generalize to N-D instead of 2-D, and also different convolution methods</sup>

[1]:https://github.com/scipy/scipy/blob/829dc206bf9630d0c4af4397c928e5770e12abe3/scipy/signal/signaltools.py#L440-L473